qpc/report/download.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2018 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+#
+"""ReportDownloadCommand is used to download all reports."""
+
+from __future__ import print_function
+
+import sys
+
+from qpc import messages, report, scan
+from qpc.clicommand import CliCommand
+from qpc.request import GET, request
+from qpc.translation import _
+from qpc.utils import (validate_write_file,
+                       write_file)
+
+from requests import codes
+
+
+# pylint: disable=too-few-public-methods
+class ReportDownloadCommand(CliCommand):
+    """Defines the report download command.
+
+    This command is for downloading all reports.
+    """
+
+    SUBCOMMAND = report.SUBCOMMAND
+    ACTION = report.DOWNLOAD
+
+    def __init__(self, subparsers):
+        """Create command."""
+        # pylint: disable=no-member
+        CliCommand.__init__(self, self.SUBCOMMAND, self.ACTION,
+                            subparsers.add_parser(self.ACTION), GET,
+                            report.REPORT_URI, [codes.ok])
+        id_group = self.parser.add_mutually_exclusive_group(required=True)
+        id_group.add_argument('--scan-job', dest='scan_job_id',
+                              metavar='SCAN_JOB_ID',
+                              help=_(messages.REPORT_SCAN_JOB_ID_HELP))
+        id_group.add_argument('--report', dest='report_id',
+                              metavar='REPORT_ID',
+                              help=_(messages.REPORT_REPORT_ID_HELP))
+        self.parser.add_argument('--output-file', dest='path', metavar='PATH',
+                                 help=_(messages.REPORT_PATH_HELP),
```

Comment:
Let's add a more descriptive message here indicating the output file is a tar.gz.